src/Faure.jl

```diff
@@ -0,0 +1,98 @@
+import Base.size, Base.getindex, Base.setindex, Base.axes
+
+# Pascal triangle -- implementation needed for Faure sequences
+struct PascalMatrix{I, M<:AbstractMatrix{I}} <: LinearAlgebra.AbstractTriangular{I, M}
+    data::M
+end
+
+function PascalMatrix(dimension::I) where I<:Integer
+    pascal = UpperTriangular(ones(I, dimension, dimension))
+	for i in 2:dimension
+        for j in 2:i
+            pascal[j, i] = pascal[j-1, i-1] + pascal[j, i-1]
+        end
+    end
+    return PascalMatrix(pascal)
+end
+
+Base.:size(x::PascalMatrix) = size(x.data)
+Base.:getindex(x::PascalMatrix, inds...) = getindex(x.data, inds...)
+Base.:setindex!(A, X::PascalMatrix, inds...) = setindex!(A, X.data, inds...)
+Base.:axes(x::PascalMatrix) = Base.:axes(x.data)
+
+function Base.:^(pascal::PascalMatrix, power::Integer)
+    result = similar(pascal.data)
+    @inbounds for idx in eachindex(pascal.data)
+        i, j = Tuple(idx)
+        result[i, j] = power^(max(j-i, 0)) * pascal.data[i, j]
```

Comment:
```suggestion
        result[idx] = power^(max(j-i, 0)) * pascal.data[idx]
```

Use the Cartesian indexing when you can.